pkg/patterns/declarative/pkg/applier/direct.go

```diff
@@ -32,16 +38,29 @@ func (d *DirectApplier) Apply(ctx context.Context,
 		Out:    os.Stdout,
 		ErrOut: os.Stderr,
 	}
-	restClient := genericclioptions.NewConfigFlags(true).WithDeprecatedPasswordFlag()
 	ioReader := strings.NewReader(manifest)
 
-	b := resource.NewBuilder(restClient)
+	restClientGetter := &staticRESTClientGetter{
+		RESTMapper: restMapper,
+		RESTConfig: restConfig,
+	}
+	b := resource.NewBuilder(restClientGetter)
```

Comment:
`Apply` has `validate` arg also, so we should add manifest validation if `validate` is `true` I think.

```suggestion
	b := resource.NewBuilder(restClientGetter)

	if validate {
		v, err := cmdutil.NewFactory(&genericclioptions.ConfigFlags{}).Validator(true)
		if err != nil {
			return err
		}
		b.Schema(v)
	}
```

I refered for initializing validator from [here](https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/apply/apply.go#L264).

Reply:
Thanks - merging.  One challenge that I'm trying to avoid is that when we build a new factory, we often have to repeat e.g. discovery against the server.

Maybe we should introduce an options struct here (1) to simplify the arguments and (2) so our API can be stable.

Reply:
Thank you to merging my suggestion.
I agree with your comment.
arguments as option is better solution, I think also.